### ISSUE TITLE
Test failures in the Arrow nightly integration builds

### ISSUE THREAD

**Issue by jorisvandenbossche** [JDASoftwareGroup/kartothek#475]
We started to see some test failures in the nightly integration tests we run in Arrow (running the kartothek tests against the latest pyarrow). Specifically, the `test_update_dataset_from_ddf_empty` test is failing:

```
=================================== FAILURES ===================================
___________________ test_update_dataset_from_ddf_empty[True] ___________________

store_factory = functools.partial(<function get_store_from_url at 0x7f1434733050>, 'hfs:///tmp/pytest-of-root/pytest-0/test_update_dataset_from_ddf_e0/store')
shuffle = True

    @pytest.mark.parametrize("shuffle", [True, False])
    def test_update_dataset_from_ddf_empty(store_factory, shuffle):
        with pytest.raises(ValueError, match="Cannot store empty datasets"):
            update_dataset_from_ddf(
>               dask.dataframe.from_delayed([], meta=(("a", int),)),
                store_factory,
                dataset_uuid="output_dataset_uuid",
                table="core",
                shuffle=shuffle,
                partition_on=["a"],
            ).compute()

tests/io/dask/dataframe/test_update.py:57: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

dfs = [], meta = (('a', <class 'int'>),), divisions = None
prefix = 'from-delayed', verify_meta = True

    @insert_meta_param_description
    def from_delayed(
        dfs, meta=None, divisions=None, prefix="from-delayed", verify_meta=True
    ):
        """Create Dask DataFrame from many Dask Delayed objects
    
        Parameters
        ----------
        dfs : list of Delayed
            An iterable of ``dask.delayed.Delayed`` objects, such as come from
            ``dask.delayed`` These comprise the individual partitions of the
            resulting dataframe.
        $META
        divisions : tuple, str, optional
            Partition boundaries along the index.
            For tuple, see https://docs.dask.org/en/latest/dataframe-design.html#partitions
            For string 'sorted' will compute the delayed values to find index
            values.  Assumes that the indexes are mutually sorted.
            If None, then won't use index information
        prefix : str, optional
            Prefix to prepend to the keys.
        verify_meta : bool, optional
            If True check that the partitions have consistent metadata, defaults to True.
        """
        from dask.delayed import Delayed
    
        if isinstance(dfs, Delayed):
            dfs = [dfs]
        dfs = [
            delayed(df) if not isinstance(df, Delayed) and hasattr(df, "key") else df
            for df in dfs
        ]
        for df in dfs:
            if not isinstance(df, Delayed):
                raise TypeError("Expected Delayed object, got %s" % type(df).__name__)
    
>       parent_meta = delayed(make_meta)(dfs[0]).compute()
E       IndexError: list index out of range

/opt/conda/envs/arrow/lib/python3.7/site-packages/dask/dataframe/io/io.py:591: IndexError
```

(from https://github.com/ursacomputing/crossbow/runs/2756067090)

Anybody know what the cause of this could be? (I don't directly see whether it's related to pyarrow, or actually a kartothek issue)

**Comment by jorisvandenbossche** [comment]
It only started to fail a bit more than a week ago, and comparing the last working with first failing build, I notice one change in the environment: dask updated from 2021.5.0 to 2021.5.1. So that might be the culprit.

Are you testing with the latest dask?

**Comment by fjetter** [comment]
> Are you testing with the latest dask?

No, only `stable`

The traceback already looks very wrong for the dask code since it tries to compute meta from dfs even if meta was provided.

**Comment by jorisvandenbossche** [comment]
What do you mean with "stable"? Latest release? Or is there a "stable" release? (note the above mentioned dask update that triggered the failure is in released versions)

**Comment by fjetter** [comment]
my terminology works like this :)
`stable`: actual release, now 2021.06.00
`latest`: main branch

we're testing actual releases / whatever `conda install dask` delivers.

This regression was introduced in https://github.com/dask/dask/pull/7586. I'm trying to fix it in https://github.com/dask/dask/pull/7769

As always, thanks for the report @jorisvandenbossche !

**Comment by jorisvandenbossche** [comment]
> we're testing actual releases / whatever conda install dask delivers.

So you are actually also getting the failures here, there was just not a recent commit on master that failed because of this (but eg the most recent open PR has the same failures: https://github.com/JDASoftwareGroup/kartothek/pull/474)

**Comment by fjetter** [comment]
Indeed. CI seems to be deeply red for a while now :/

**Comment by jorisvandenbossche** [comment]
And thanks for fixing it on the dask side!

**Comment by steffen-schroeder-by** [comment]
Fixed the CI by excluding the 2 dask versions which contain the issue - assuming that `>2020.6.0` will contain the fix dask/dask#7769

**Comment by steffen-schroeder-by** [comment]
Thanks for reporting @jorisvandenbossche - really appreciated 👍
